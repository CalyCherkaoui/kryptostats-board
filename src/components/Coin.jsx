/* eslint-disable no-console */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { IconContext } from 'react-icons';
import { FaAngleUp, FaAngleDown } from 'react-icons/fa';
import { GrAddCircle } from 'react-icons/gr';
import { AiOutlineCloseCircle } from 'react-icons/ai';

const Coin = ({
  coin,
  indexList,
  deleteCoin,
  addCoin,
}) => {
  const upDownIconRender = () => {
    if (coin.price_change_percentage_24h >= 0) {
      return (
        <IconContext.Provider value={{ color: 'green', className: 'up_icon' }}>
          <FaAngleUp />
        </IconContext.Provider>
      );
    }
    return (
      <IconContext.Provider value={{ color: 'red', className: 'down_icon' }}>
        <FaAngleDown />
      </IconContext.Provider>
    );
  };

  const btnIconRender = () => {
    if (indexList) {
      return (
        <button
          className="coinlist_item_btn"
          type="button"
          onClick={
            (e) => {
              e.preventDefault();
              addCoin(coin.id);
              console.log('added');
            }
          }
        >
          <IconContext.Provider value={{ color: 'green', className: 'add_icon' }}>
            <GrAddCircle />
          </IconContext.Provider>
        </button>
      );
    }
    return (
      <button
        className="coinlist_item_btn"
        type="button"
        onClick={
          (e) => {
            e.preventDefault();
            deleteCoin(coin.id);
            console.log('deleted');
          }
        }
      >
        <IconContext.Provider value={{ color: 'red', className: 'close_icon' }}>
          <AiOutlineCloseCircle />
        </IconContext.Provider>
      </button>
    );
  };
  return (
    <Link to={`/coininfo/${coin.id}`}>
      <li className="coinlist_item">
        <img src={coin.image} alt={coin.name} className="coinlist_item_image" />
        <span className="coinlist_item_price">{coin.current_price}</span>
        <span className="coinlist_item_change">
          <i className="upDown_icon">
            { upDownIconRender()}
          </i>
          {coin.price_change_percentage_24h}
        </span>
        { btnIconRender() }
      </li>
    </Link>
  );
};

Coin.propTypes = {
  coin: PropTypes.shape(
    {
      id: PropTypes.number,
      name: PropTypes.string,
      image: PropTypes.string,
      current_price: PropTypes.number,
      price_change_percentage_24h: PropTypes.number,
    },
  ).isRequired,
  indexList: PropTypes.bool.isRequired,
  deleteCoin: PropTypes.func.isRequired,
  addCoin: PropTypes.func.isRequired,
};

export default Coin;
