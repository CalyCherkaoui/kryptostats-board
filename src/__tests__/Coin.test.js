import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Coin from '../components/Coin';

describe(
  'Renders correctly the Coin component',
  () => {
    let coin;
    beforeEach(() => {
      coin = {
        id: 'testcoin',
        image: 'image_link',
        current_price: 1,
        price_change_percentage_24h: 1,
      };
    });

    it(
      'Matches Coin snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <Coin coin={coin} />
            </Router>
          </Provider>,
        );
        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
