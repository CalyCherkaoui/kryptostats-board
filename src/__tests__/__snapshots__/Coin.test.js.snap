// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly the Coin component Matches Coin snapshot 1`] = `
<a
  className="coin_wrapper shadowed_small flex_space_even"
  href="/coininfo/testcoin"
  onClick={[Function]}
>
  <img
    className="coinlist_item_image"
    src="image_link"
  />
  <div
    className="coin_info_wrapper flex_col flex_center"
  >
    <div
      className="coinlist_item_info flex_col flex_center"
    >
      <div
        className="coin_legend normal_typography"
      >
        Price:
      </div>
      <div
        className="coin_data spaced_typography"
      >
        1
      </div>
    </div>
    <div
      className="coinlist_item_info flex_col flex_center"
    >
      <div
        className="coin_legend normal_typography"
      >
        Daily Change:
      </div>
      <div
        className="coin_data spaced_typography"
      >
        <i
          className="updouwn_icon"
        >
          <svg
            className="up_icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "green",
              }
            }
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
            />
          </svg>
        </i>
        1
      </div>
    </div>
  </div>
</a>
`;
